### ISSUE TITLE
Support passing filechooser instance as an argument to the callback function

### ISSUE THREAD

**Issue by ptooley** [crahan/ipyfilechooser#13]
Callback functionality would be greatly increased by the callback function being able to access the calling instance, in my use case I am using multiple file choosers to allow users to add multiple files.  

This can be added without breaking existing callback usage by, for example:

```
try:
    self._callback(self)
except TypeError:
    self._callback()
```

**Comment by crahan** [comment]
@ptooley can you share a code example of what you're trying to do? Major spoilers: I'm no Python expert by any means. I'd love to add what you're asking but I'd like to better understand. Thanks!

**Comment by ptooley** [comment]
No problem.  What I'd like to be able to do is access the properties of the updated filechooser from within the callback function, so for example:

```python
class CustomGuiWidget(widgets.VBox):

def __init__(self, n_files, *args, **kwargs):
    self._msgbox = HBox()
    self._choosers = [FileChooser()]*n_loaders
    self._files = {chooser: None for chooser in self._choosers}
    [chooser.register_callback(self.validate_file_callback) for chooser in self._choosers]

    self._setup_msgbox()
    super().__init__(children=[self._loader, self._msgbox], *args, **kwargs)

def validate_file_callback(chooser):
    try:
        check_file_valid(chooser.selected)
        self.show_message("File {} valid".format(chooser.selected))
        self._files[chooser] = chooser.selected
        self._files_updated_hook()
    except WrongFileType:
        self.show_error("File {} is not a valid foofile!")

def _setup_msgbox():
    # Widget magic here....

# class def continues...
```

So here I have multiple filechoosers that the user will want to update at various points.  I can also dynamically add and remove choosers.  Having the chooser pass itself as a callback argument lets me directly deal with the updated object, rather than for me to have to parse through a list of them all and try to work out which one has changed.  This is particularly useful since ``FileChooser`` is a hashable type so I can use it directly as a dict key too :)

**Comment by crahan** [comment]
Oh, that makes total sense. I'll get that PR approved asap and push out a new version.

**Comment by crahan** [comment]
Thanks again for the pull request! I've merged https://github.com/crahan/ipyfilechooser/pull/14, updated the documentation, and published v0.3.3 to PyPi.
